src/commands/server.cpp

```diff
@@ -22,6 +22,61 @@
 #include <vector>
 
 volatile std::sig_atomic_t eoserv_sig_abort = false;
+volatile std::sig_atomic_t eoserv_sig_reload = false;
+
+TimeEvent* shutdown_timer = nullptr;
+
+static void schedule_shutdown(const std::vector<std::string>& arguments, Command_Source* from, bool is_shutdown)
+{
+	if (shutdown_timer != nullptr)
+	{
+		from->ServerMsg("Shutdown/reload is already scheduled. Use $cancel to cancel pending shutdown/reload.");
+		return;
+	}
+
+	int timeout = 0;
+	if (arguments.size() >= 1)
+	{
+		if (util::lowercase(arguments[0]) != "now")
+		{
+			timeout = util::variant(arguments[0]).GetInt();
+			if (timeout > 0)
+			{
+				char printBuf[250] = {0};
+				snprintf(printBuf, 250, "Attention!! Server will be %s in %d seconds", is_shutdown ? "shut down" : "reloaded", timeout);
+				from->SourceWorld()->ServerMsg(std::string(printBuf));
+			}
+			else
+			{
+				timeout = 0;
+			}
+		}
+	}
+
+	// capturing lambdas cannot be passed as function pointers, therefore is_shutdown cannot be captured
+	TimerCallback shutdown_callback = nullptr;
+	if (is_shutdown)
+	{
+		shutdown_callback = [](void * input)
+		{
+			Command_Source* input_from = static_cast<Command_Source*>(input);
+			Console::Wrn("Server shut down by %s", input_from->SourceName().c_str());
```

Comment:
I know that this is a warning in EOSERV, but should it be?
Shutting down the server is just a part of regular operation. Would `Console::Out` be more appropriate?

Reply:
I don't have a strong feeling about it either way. Maybe it's a warning due to the absence of an admin log. I can change it to `Out`.